### PR TITLE
AV-167750 : Shared VIP support in AKO for NodePortLocal mode with Ant…

### PR DESCRIPTION
**Pull request by DixitAakash**
Shared VIP support was lacking in AKO for NodePortLocal mode with Antrea CNI. As part of this PR, Shared VIP support will be added for NPL deployments with Antrea CNI. The test results are given below :
```
+----------------------------------------------------------------------------------+----------------+---------------------+-------------+-------------+
| Absolute Path:Test case                                                          |     Results    |       Duration      |  Start Time |   End Time  |
+----------------------------------------------------------------------------------+----------------+---------------------+-------------+-------------+
|      Test Suite Path and Name : avi-                                             |                |                     |             |             |
| dev/test/avitest/functional/ako/test_lbsvc.py                                    |                |                     |             |             |
| ++++++++++++++++++++++++++++++++++++++++                                         |                |                     |             |             |
| Test_LBSvcSNI_NonDedicated_Shard.test_app_creation                               |     PASSED     |       00:00:43      |   21:31:48  |   21:32:32  |
| Test_LBSvcSNI_NonDedicated_Shard.test_update_service                             |     PASSED     |       00:00:27      |   21:32:32  |   21:32:59  |
| Test_LBSvcSNI_NonDedicated_Shard.test_update_deployment_replicas                 |     PASSED     |       00:00:42      |   21:32:59  |   21:33:41  |
| Test_LBSvcSNI_NonDedicated_Shard.test_create_multiple_services                   |     PASSED     |       00:00:14      |   21:33:41  |   21:33:56  |
| Test_LBSvcSNI_NonDedicated_Shard.test_delete_multiple_services                   |     PASSED     |       00:00:01      |   21:33:56  |   21:33:57  |
| Test_LBSvcSNI_NonDedicated_Shard.test_app_no_pod                                 |     PASSED     |       00:00:15      |   21:33:57  |   21:34:13  |
| Test_LBSvcSNI_NonDedicated_Shard.test_service_static_ip                          |     PASSED     |       00:01:03      |   21:34:13  |   21:35:16  |
| Test_LBSvcSNI_NonDedicated_Shard.test_ako_delete_pod_with_vs_deletion_static_ip  |     PASSED     |       00:00:32      |   21:35:16  |   21:35:48  |
| Test_LBSvcSNI_NonDedicated_Shard.test_ako_auto_fqdn                              |     PASSED     |       00:00:56      |   21:35:48  |   21:36:44  |
| Test_LBSvcSNI_NonDedicated_Shard.test_ako_fqdn_annotation                        |     PASSED     |       00:00:38      |   21:36:44  |   21:37:23  |
| Test_LBSvcSNI_NonDedicated_Shard.test_svc_infrasetting_update_segroup            |    SKIPPED     |       00:00:00      |   21:37:23  |   21:37:23  |
| Test_LBSvcSNI_NonDedicated_Shard.test_svc_infrasetting_update_network            |     PASSED     |       00:00:53      |   21:37:23  |   21:38:17  |
| Test_LBSvcSNI_NonDedicated_Shard.test_public_ip_allocation                       |    SKIPPED     |       00:00:00      |   21:38:17  |   21:38:17  |
| Test_LBSvcSNI_NonDedicated_Shard.test_create_multiple_services_shared_vip        |     PASSED     |       00:00:27      |   21:38:17  |   21:38:45  |
| Test_LBSvcSNI_NonDedicated_Shard.test_create_multiple_services_shared_vip_static |     PASSED     |                     |             |             |
| _ip                                                                              |                |       00:00:23      |   21:38:45  |   21:39:08  |
| Test_LBSvcSNI_NonDedicated_Shard.test_app_creation_sctp                          |    SKIPPED     |       00:00:00      |   21:39:08  |   21:39:08  |
| Test_LBSvcSNI_NonDedicated_Shard.test_create_multiple_services_shared_vip_sctp   |    SKIPPED     |       00:00:00      |   21:39:08  |   21:39:08  |
| Test_LBSvcSNI_NonDedicated_Shard.test_delete_apps                                |     PASSED     |       00:00:00      |   21:39:08  |   21:39:08  |
| -------------------------------------------------------------------------------- |  ------------  |     -----------     | ----------- | ----------- |
|                                                                                  |   Passed: 14   |                     |             |             |
|                                                                                  |    Failed:0    |                     |             |             |
|  Total test cases: 18                                                            |    Skipped:4   | Total Time:00:07:19 |             |             |
+----------------------------------------------------------------------------------+----------------+---------------------+-------------+-------------+


+----------------------------------------------------------------------------------+----------------+---------------------+-------------+-------------+
| Absolute Path:Test case                                                          |     Results    |       Duration      |  Start Time |   End Time  |
+----------------------------------------------------------------------------------+----------------+---------------------+-------------+-------------+
|      Test Suite Path and Name : avi-                                             |                |                     |             |             |
| dev/test/avitest/functional/ako/test_lbsvc.py                                    |                |                     |             |             |
| ++++++++++++++++++++++++++++++++++++++++                                         |                |                     |             |             |
| Test_LBSvcSNI_Dedicated_Shard.test_app_creation                                  |     PASSED     |       00:00:27      |   21:41:35  |   21:42:03  |
| Test_LBSvcSNI_Dedicated_Shard.test_update_service                                |     PASSED     |       00:00:27      |   21:42:03  |   21:42:31  |
| Test_LBSvcSNI_Dedicated_Shard.test_update_deployment_replicas                    |     PASSED     |       00:00:32      |   21:42:31  |   21:43:03  |
| Test_LBSvcSNI_Dedicated_Shard.test_create_multiple_services                      |     PASSED     |       00:00:44      |   21:43:03  |   21:43:48  |
| Test_LBSvcSNI_Dedicated_Shard.test_delete_multiple_services                      |     PASSED     |       00:00:01      |   21:43:48  |   21:43:49  |
| Test_LBSvcSNI_Dedicated_Shard.test_app_no_pod                                    |     PASSED     |       00:00:15      |   21:43:49  |   21:44:05  |
| Test_LBSvcSNI_Dedicated_Shard.test_service_static_ip                             |     PASSED     |       00:01:28      |   21:44:05  |   21:45:33  |
| Test_LBSvcSNI_Dedicated_Shard.test_ako_delete_pod_with_vs_deletion_static_ip     |     PASSED     |       00:00:32      |   21:45:33  |   21:46:05  |
| Test_LBSvcSNI_Dedicated_Shard.test_ako_auto_fqdn                                 |     PASSED     |       00:01:06      |   21:46:05  |   21:47:12  |
| Test_LBSvcSNI_Dedicated_Shard.test_ako_fqdn_annotation                           |     PASSED     |       00:00:38      |   21:47:12  |   21:47:51  |
| Test_LBSvcSNI_Dedicated_Shard.test_svc_infrasetting_update_segroup               |    SKIPPED     |       00:00:00      |   21:47:51  |   21:47:51  |
| Test_LBSvcSNI_Dedicated_Shard.test_svc_infrasetting_update_network               |     PASSED     |       00:00:53      |   21:47:51  |   21:48:44  |
| Test_LBSvcSNI_Dedicated_Shard.test_public_ip_allocation                          |    SKIPPED     |       00:00:00      |   21:48:44  |   21:48:44  |
| Test_LBSvcSNI_Dedicated_Shard.test_create_multiple_services_shared_vip           |     PASSED     |       00:00:12      |   21:48:44  |   21:48:57  |
| Test_LBSvcSNI_Dedicated_Shard.test_create_multiple_services_shared_vip_static_ip |     PASSED     |       00:00:44      |   21:48:57  |   21:49:41  |
| Test_LBSvcSNI_Dedicated_Shard.test_app_creation_sctp                             |    SKIPPED     |       00:00:00      |   21:49:41  |   21:49:41  |
| Test_LBSvcSNI_Dedicated_Shard.test_create_multiple_services_shared_vip_sctp      |    SKIPPED     |       00:00:00      |   21:49:41  |   21:49:41  |
| Test_LBSvcSNI_Dedicated_Shard.test_delete_apps                                   |     PASSED     |       00:00:00      |   21:49:41  |   21:49:42  |
| -------------------------------------------------------------------------------- |  ------------  |     -----------     | ----------- | ----------- |
|                                                                                  |   Passed: 14   |                     |             |             |
|                                                                                  |    Failed:0    |                     |             |             |
|  Total test cases: 18                                                            |    Skipped:4   | Total Time:00:08:06 |             |             |
+----------------------------------------------------------------------------------+----------------+---------------------+-------------+-------------+


+----------------------------------------------------------------------------------+----------------+---------------------+-------------+-------------+
| Absolute Path:Test case                                                          |     Results    |       Duration      |  Start Time |   End Time  |
+----------------------------------------------------------------------------------+----------------+---------------------+-------------+-------------+
|      Test Suite Path and Name : avi-                                             |                |                     |             |             |
| dev/test/avitest/functional/ako/test_lbsvc.py                                    |                |                     |             |             |
| ++++++++++++++++++++++++++++++++++++++++                                         |                |                     |             |             |
| Test_LBSvcEVH_NonDedicated_Shard.test_app_creation                               |     PASSED     |       00:00:12      |   21:50:47  |   21:51:00  |
| Test_LBSvcEVH_NonDedicated_Shard.test_update_service                             |     PASSED     |       00:00:37      |   21:51:00  |   21:51:37  |
| Test_LBSvcEVH_NonDedicated_Shard.test_update_deployment_replicas                 |     PASSED     |       00:00:42      |   21:51:37  |   21:52:20  |
| Test_LBSvcEVH_NonDedicated_Shard.test_create_multiple_services                   |     PASSED     |       00:00:15      |   21:52:20  |   21:52:35  |
| Test_LBSvcEVH_NonDedicated_Shard.test_delete_multiple_services                   |     PASSED     |       00:00:01      |   21:52:35  |   21:52:37  |
| Test_LBSvcEVH_NonDedicated_Shard.test_app_no_pod                                 |     PASSED     |       00:00:15      |   21:52:37  |   21:52:52  |
| Test_LBSvcEVH_NonDedicated_Shard.test_service_static_ip                          |     PASSED     |       00:01:03      |   21:52:52  |   21:53:55  |
| Test_LBSvcEVH_NonDedicated_Shard.test_ako_delete_pod_with_vs_deletion_static_ip  |     PASSED     |       00:00:47      |   21:53:55  |   21:54:43  |
| Test_LBSvcEVH_NonDedicated_Shard.test_ako_auto_fqdn                              |     PASSED     |       00:00:55      |   21:54:43  |   21:55:39  |
| Test_LBSvcEVH_NonDedicated_Shard.test_ako_fqdn_annotation                        |     PASSED     |       00:00:39      |   21:55:39  |   21:56:18  |
| Test_LBSvcEVH_NonDedicated_Shard.test_svc_infrasetting_update_segroup            |    SKIPPED     |       00:00:00      |   21:56:18  |   21:56:18  |
| Test_LBSvcEVH_NonDedicated_Shard.test_svc_infrasetting_update_network            |     PASSED     |       00:00:53      |   21:56:18  |   21:57:12  |
| Test_LBSvcEVH_NonDedicated_Shard.test_public_ip_allocation                       |    SKIPPED     |       00:00:00      |   21:57:12  |   21:57:12  |
| Test_LBSvcEVH_NonDedicated_Shard.test_create_multiple_services_shared_vip        |     PASSED     |       00:00:27      |   21:57:12  |   21:57:40  |
| Test_LBSvcEVH_NonDedicated_Shard.test_create_multiple_services_shared_vip_static |     PASSED     |                     |             |             |
| _ip                                                                              |                |       00:00:23      |   21:57:40  |   21:58:03  |
| Test_LBSvcEVH_NonDedicated_Shard.test_app_creation_sctp                          |    SKIPPED     |       00:00:00      |   21:58:03  |   21:58:03  |
| Test_LBSvcEVH_NonDedicated_Shard.test_create_multiple_services_shared_vip_sctp   |    SKIPPED     |       00:00:00      |   21:58:03  |   21:58:03  |
| Test_LBSvcEVH_NonDedicated_Shard.test_delete_apps                                |     PASSED     |       00:00:00      |   21:58:03  |   21:58:04  |
| -------------------------------------------------------------------------------- |  ------------  |     -----------     | ----------- | ----------- |
|                                                                                  |   Passed: 14   |                     |             |             |
|                                                                                  |    Failed:0    |                     |             |             |
|  Total test cases: 18                                                            |    Skipped:4   | Total Time:00:07:16 |             |             |
+----------------------------------------------------------------------------------+----------------+---------------------+-------------+-------------+


+----------------------------------------------------------------------------------+----------------+---------------------+-------------+-------------+
| Absolute Path:Test case                                                          |     Results    |       Duration      |  Start Time |   End Time  |
+----------------------------------------------------------------------------------+----------------+---------------------+-------------+-------------+
|      Test Suite Path and Name : avi-                                             |                |                     |             |             |
| dev/test/avitest/functional/ako/test_lbsvc.py                                    |                |                     |             |             |
| ++++++++++++++++++++++++++++++++++++++++                                         |                |                     |             |             |
| Test_LBSvcEVH_Dedicated_Shard.test_app_creation                                  |     PASSED     |       00:00:43      |   14:14:38  |   14:15:22  |
| Test_LBSvcEVH_Dedicated_Shard.test_update_service                                |     PASSED     |       00:00:37      |   14:15:22  |   14:15:59  |
| Test_LBSvcEVH_Dedicated_Shard.test_update_deployment_replicas                    |     PASSED     |       00:00:32      |   14:15:59  |   14:16:31  |
| Test_LBSvcEVH_Dedicated_Shard.test_create_multiple_services                      |     PASSED     |       00:00:14      |   14:16:31  |   14:16:46  |
| Test_LBSvcEVH_Dedicated_Shard.test_delete_multiple_services                      |     PASSED     |       00:00:01      |   14:16:46  |   14:16:47  |
| Test_LBSvcEVH_Dedicated_Shard.test_app_no_pod                                    |     PASSED     |       00:00:15      |   14:16:47  |   14:17:03  |
| Test_LBSvcEVH_Dedicated_Shard.test_service_static_ip                             |     PASSED     |       00:01:18      |   14:17:03  |   14:18:21  |
| Test_LBSvcEVH_Dedicated_Shard.test_ako_delete_pod_with_vs_deletion_static_ip     |     PASSED     |       00:00:32      |   14:18:21  |   14:18:53  |
| Test_LBSvcEVH_Dedicated_Shard.test_ako_auto_fqdn                                 |     PASSED     |       00:01:06      |   14:18:53  |   14:19:59  |
| Test_LBSvcEVH_Dedicated_Shard.test_ako_fqdn_annotation                           |     PASSED     |       00:00:38      |   14:19:59  |   14:20:38  |
| Test_LBSvcEVH_Dedicated_Shard.test_svc_infrasetting_update_segroup               |    SKIPPED     |       00:00:00      |   14:20:38  |   14:20:38  |
| Test_LBSvcEVH_Dedicated_Shard.test_svc_infrasetting_update_network               |     PASSED     |       00:00:53      |   14:20:38  |   14:21:32  |
| Test_LBSvcEVH_Dedicated_Shard.test_public_ip_allocation                          |    SKIPPED     |       00:00:00      |   14:21:32  |   14:21:32  |
| Test_LBSvcEVH_Dedicated_Shard.test_create_multiple_services_shared_vip           |     PASSED     |       00:00:12      |   14:21:32  |   14:21:45  |
| Test_LBSvcEVH_Dedicated_Shard.test_create_multiple_services_shared_vip_static_ip |     PASSED     |       00:00:43      |   14:21:45  |   14:22:29  |
| Test_LBSvcEVH_Dedicated_Shard.test_app_creation_sctp                             |    SKIPPED     |       00:00:00      |   14:22:29  |   14:22:29  |
| Test_LBSvcEVH_Dedicated_Shard.test_create_multiple_services_shared_vip_sctp      |    SKIPPED     |       00:00:00      |   14:22:29  |   14:22:29  |
| Test_LBSvcEVH_Dedicated_Shard.test_delete_apps                                   |     PASSED     |       00:00:00      |   14:22:29  |   14:22:30  |
| -------------------------------------------------------------------------------- |  ------------  |     -----------     | ----------- | ----------- |
|                                                                                  |   Passed: 14   |                     |             |             |
|                                                                                  |    Failed:0    |                     |             |             |
|  Total test cases: 18                                                            |    Skipped:4   | Total Time:00:07:51 |             |             |
+----------------------------------------------------------------------------------+----------------+---------------------+-------------+-------------+
```

### CODE REVIEW

**Comment by DixitAakash** [comment]
build ako

File: internal/nodes/dequeue_ingestion.go [hunk 1]
```diff
@@ -382,13 +378,28 @@ func handlePod(key, namespace, podName string, fullsync bool) {
...
+		utils.AviLog.Infof("key: %s, msg: NPL Services retrieved: %s", key, services)
+	}
+}
+
+func handleLBSvcUpdateForPod(key string, lbSvcs []string, fullsync bool) {
+	for _, lbSvc := range lbSvcs {
+		lbSvcSplit := strings.Split(lbSvc, "/")
+		lbSvcNS, lbSvcName := lbSvcSplit[0], lbSvcSplit[1]
+		sharedVipLBSvcKey := lib.SharedVipServiceKey + "/" + lbSvc
+		sharedVipKeys, keysFound := ServiceChanges(lbSvcName, lbSvcNS, sharedVipLBSvcKey)
```
Comment (akshayhavile) [inline internal/nodes/dequeue_ingestion.go:392]: @DixitAakash: We can ignore `keysFound` as function always true.  Also we can take out ` utils.CheckIfNamespaceAccepted(lbSvcNS) ` as all lbsvc are from accepted namespace ( line number 358)

Comment (DixitAakash) [reply internal/nodes/dequeue_ingestion.go]: Done

**Review by akshayhavile — APPROVED** [review]

**Comment by DixitAakash** [comment]
build ako

**Comment by DixitAakash** [comment]
build ako

**Comment by DixitAakash** [comment]
build ako

**Comment by DixitAakash** [comment]
build ako

**Review by swathinsankaran — APPROVED** [review]

**Comment by swathinsankaran** [comment]
@akshayhavile, @DixitAakash We have to document this, right?

**Comment by DixitAakash** [comment]
build ako

**Comment by DixitAakash** [comment]
> @akshayhavile, @DixitAakash We have to document this, right?
Do we need a separate document for this? The existing document here https://avinetworks.com/docs/ako/1.8/shared-vip/ does not say anything about not supporting Shared vip for any particular service type.

**Pull request merged**